compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -340,7 +341,9 @@ generateSrcFromLf (Qualify qualify) thisPkgId pkgMap m = noLoc mod
                                   , dd_cType = Nothing
                                   , dd_kindSig = Nothing
                                   , dd_cons = convDataCons dataTypeCon0 dataCons
-                                  , dd_derivs = noLoc []
+                                  , dd_derivs = noLoc
+                                        [ noLoc $ HsDerivingClause noExt Nothing $ noLoc $
+                                          map (HsIB noExt . mkClsTy) ["Eq", "Show"] ]
```

Comment:
Added deriving clause to generated template data type decls. I suspect the "dummy" template instance just below may fail however, if the type arguments do not have `Eq` and `Show` constraints.